### ISSUE TITLE
cannot download Git auth OTS: certificate signed by unknown authority location

### ISSUE THREAD

**Issue by jgallucci32** [gitpod-io/gitpod#2050]
### Describe the bug

When building a custom image using a Dockerfile in your .gitpod.yml configuration, it fails to build if Gitpod is using an SSL certificate signed by a custom CA with the following error:
```
initializing base image context
cannot download OTS                           attempt=0 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
cannot download OTS                           attempt=1 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
cannot download OTS                           attempt=2 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
cannot download OTS                           attempt=3 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
cannot download OTS                           attempt=4 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
cannot download OTS                           attempt=5 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
cannot download OTS                           attempt=6 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
cannot download OTS                           attempt=7 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
cannot download OTS                           attempt=8 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
acannot download OTS                           attempt=9 error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority
cannot download Git auth OTS                  error=Get "https://gitpod.domain.local/api/ots/get/995f444b-4e38-4c90-98d4-d0dc618b4a3a": x509: certificate signed by unknown authority location=/workspace/init
init failed                                   error=cannot initialize workspace: git initializer: rpc error: code = InvalidArgument desc = cannot get OTS
init failed                                   error=cannot initialize workspace: git initializer: rpc error: code = InvalidArgument desc = cannot get OTS

build failed: cannot run context initializer: container exited with non-zero exit code: 1init failed                                   error=cannot initialize workspace: git initializer:rpc error: code = InvalidArgument desc = cannot get OTS

Error: build failed: cannot run context initializer: container exited with non-zero exit code: 1
```
The reason this is happening is the selfbuild image is hard-coded to use `alpine:3.9` here https://github.com/gitpod-io/gitpod/blob/1e1b01c0529c31ad7e9c1e60315d887abe741791/components/image-builder/pkg/builder/selfbuild.go#L25

Since Alpine does not know about the custom CA certificate there is no way for it to trust it. This means the selfbuild image also does not trust it and there is nothing in the deployment which allows you to add the certificate.

### Steps to reproduce

1. Deploy Gitpod self-hosted using a custom SSL certificate
2. Configure a Gitpod project to use a custom Dockerfile
3. Launch workspace

### Expected behavior

Workspace builds custom docker image and launches workspace.

### Additional information

Red Hat Enterprise Linux 7.8
Docker CE 19.03.13
Gitpod v0.5.0

**Comment by jgallucci32** [comment]
This is one which could not be fixed with a simple Helm chart modification like other SSL cert problems since the selfbuild containers are launched outside of Kubernetes using Docker-in-Docker. Therefore a sidecar must be created to fix this as a workaround. Steps for the workaround:

1. Copy `ca-bundle.crt` to the docker build directory
2. Create the `fix.sh` script
```bash
#!/bin/sh
echo "Fixing selfbuild image SSL root certificates."

# Get selfbuild image information
selfBuildImageId=$(docker -H tcp://127.0.0.1:2375 images -q --filter=reference='gitpod.io/image-builder/selfbuild:*')
selfBuildImageName=$(docker -H tcp://127.0.0.1:2375 images --filter=reference='gitpod.io/image-builder/selfbuild:*' --format "{{.Repository}}:{{.Tag}}")

# Backup existing selfbuild image
docker -H tcp://127.0.0.1:2375 tag $selfBuildImageId selfbuild-backup

# Create Dockerfile
cat << EOF > Dockerfile
FROM $selfBuildImageId
COPY ca-bundle.crt /etc/ssl/certs/ca-certificates.crt
RUN /bin/ln -sf /etc/ssl/certs/ca-certificates.crt /etc/ssl/cert.pem
EOF

# Build image
docker -H tcp://127.0.0.1:2375 build . -t $selfBuildImageName

# Keep container running
tail -f /dev/null
```
3. Create the `Dockerfile`
```
FROM docker:18.06-dind

WORKDIR /build
COPY ca-bundle.crt /build/ca-bundle.crt
COPY fix.sh /run/fix.sh

RUN chmod +x /run/fix.sh
ENTRYPOINT ["/run/fix.sh"]
```
4. Build and push image
```
docker build -t registry.domain.local/gitpod/fix-image-builder .
docker push registry.domain.local/gitpod/fix-image-builder
```
5. Add the sidecar to the `image-builder` pod with the following manifest
```yaml
      - image: registry.domain.local/gitpod/fix-image-builder
        imagePullPolicy: Always
        name: fix-image-builder
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
```

**Comment by jgallucci32** [comment]
I can confirm this is still an issue in 0.6.0, however, the workaround is still valid so we'll keep using the sidecar method until this is addressed.

**Comment by jgallucci32** [comment]
UPDATE: Looks like this is resolved with changes in #3198 as the customer can now specify a custom Alpine image with updated SSL certificates.
